pandas/core/ops/__init__.py

```diff
@@ -1702,10 +1703,16 @@ def wrapper(left, right):
             #  does inference in the case where `result` has object-dtype.
             return construct_result(left, result, index=left.index, name=res_name)
 
+        elif isinstance(right, (ABCDatetimeArray, pd.DatetimeIndex)):
+            result = op(left._values, right)
+            return construct_result(left, result, index=left.index, name=res_name)
+
         lvalues = left.values
         rvalues = right
         if isinstance(rvalues, ABCSeries):
```

Comment:
these cases are the same, yes? 

``isinstance(rvalues, (ABCSeries, ABCIndexClass))``